sqlitecluster/SQLite.cpp

```diff
@@ -74,7 +74,9 @@ SQLite::SQLite(const string& filename, int cacheSize, int autoCheckpoint, int ma
     SASSERT(!SQuery(_db, "new file format for DESC indexes", "PRAGMA legacy_file_format = OFF"));
 
     // These other pragmas only relate to read/write databases.
-    SASSERT(!SQuery(_db, "disabling synchronous commits", "PRAGMA synchronous = OFF;"));
+    string synchronousLog = (synchronous == "false" ? "disabling" : "enabling") + " synchronous commits";
```

Comment:
This is trying to just add two `const char*` pointers, so probably won't work.  I'd suggest casting the result of the first parenthesis to `(string)` to convert the pointer to an STL string.  Then the expression is adding a `string` to a `const char*` which should work as you like it.

HOWEVER, I think a confusion here is in thinking `synchronous` is true/false.  Here are the actual string options: https://sqlite.org/pragma.html#pragma_synchronous  So rather than assuming it's enable/disable, just log what the new setting is without worrying about interpreting the result into an "enable/disable" message.

(Furthermore, for future reference -- when doing string comparisons, please use `SIEquals()` for a case-insensitive compare.  However, I still think you shouldn't compare it to anything and should instead just log the value requested.)

Reply:
👍 that is similar to what Cole suggested 1:1. 

> except then you'll have issues down stream.
> `SQuery` takes `db, char, string`
> i would skip on making the log line say what we're setting it to
> because we're going to log that when we run the SQuery
> so you can just leave the `synchronousLog` as `setting setting database synchronicity` or something

Thanks for the tips though!